### PR TITLE
Fixes #18326: Handle proxies in rudder package command

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/18326

### CODE REVIEW

File: relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkgUtils.py [hunk 1]
```diff
@@ -118,6 +119,43 @@ def dictToAsciiTable(data):
...
+        if self.last_nonce:
+            r.headers['Proxy-Authorization'] = self.build_digest_header(r.method, r.url)
+        r.register_hook('response', self.handle_407)
+        return r
+
+def getRequest(url, stream):
+  if (PROXY_URL == ""):
+    return requests.get(completeUrl, auth=(USERNAME, PASSWORD), stream=stream)
+  else:
+    proxies = { "https": PROXY_URL }
```
Comment (peckpeck) [inline relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkgUtils.py:152]: shouldn't we handle http too ?

Comment (VinceMacBuche) [reply relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkgUtils.py:152]: I think all request are made through https but maybe yes

espcially that it's authenticated requests with password that would leak the password

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Review by peckpeck — APPROVED** [review]

**Comment by VinceMacBuche** [comment]
OK, squash merging this PR

**Pull request merged**